cmd/kubeadm/app/util/endpoint.go

```diff
@@ -54,10 +64,21 @@ func GetMasterHostPort(cfg *kubeadmapi.MasterConfiguration) (string, error) {
 		masterIP = ip.String()
 	}
 
-	if cfg.API.BindPort < 0 || cfg.API.BindPort > 65535 {
-		return "", fmt.Errorf("api server port must be between 0 and 65535")
+	var port int32
+	if len(portStr) > 0 {
+		portInt, err := strconv.Atoi(portStr)
+		if err != nil {
+			return "", fmt.Errorf("error parsing `ControlPlaneEndpoint` port value `%s`: %s", portStr, err.Error())
+		}
+		port = int32(portInt)
```

Comment:
This overrides the config.API.BindPort, are there any other areas that might conflict?  

Also, we should surface a warning on that condition so it's clear from a UX perspective.

Reply:
The `GetMasterHostPort` function is only used for generating kubeconfigs. Since `cfg.API.ControlPlaneEndpoint` already overrides `cfg.API.AdvertiseAddress` for that purpose, it seems right to me that it should also be possible to override the port.